stl/src/syserror.cpp

```diff
@@ -24,90 +24,103 @@ namespace {
+        {ERROR_OPEN_FILES, errc::device_or_resource_busy},
+        {ERROR_DEVICE_IN_USE, errc::device_or_resource_busy},
+        {ERROR_REPARSE_TAG_INVALID, errc::invalid_argument},
+        {WSAEINTR, errc::interrupted},
         {WSAEBADF, errc::bad_file_descriptor},
-        {WSAECONNABORTED, errc::connection_aborted},
-        {WSAECONNREFUSED, errc::connection_refused},
-        {WSAECONNRESET, errc::connection_reset},
-        {WSAEDESTADDRREQ, errc::destination_address_required},
+        {WSAEACCES, errc::permission_denied},
         {WSAEFAULT, errc::bad_address},
-        {WSAEHOSTUNREACH, errc::host_unreachable},
-        {WSAEINPROGRESS, errc::operation_in_progress},
-        {WSAEINTR, errc::interrupted},
         {WSAEINVAL, errc::invalid_argument},
-        {WSAEISCONN, errc::already_connected},
         {WSAEMFILE, errc::too_many_files_open},
+        {WSAEWOULDBLOCK, errc::operation_would_block},
+        {WSAEINPROGRESS, errc::operation_in_progress},
+        {WSAEALREADY, errc::connection_already_in_progress},
+        {WSAENOTSOCK, errc::not_a_socket},
+        {WSAEDESTADDRREQ, errc::destination_address_required},
         {WSAEMSGSIZE, errc::message_size},
-        {WSAENAMETOOLONG, errc::filename_too_long},
+        {WSAEPROTOTYPE, errc::wrong_protocol_type},
+        {WSAENOPROTOOPT, errc::no_protocol_option},
+        {WSAEPROTONOSUPPORT, errc::protocol_not_supported},
+        {WSAEOPNOTSUPP, errc::operation_not_supported},
+        {WSAEAFNOSUPPORT, errc::address_family_not_supported},
+        {WSAEADDRINUSE, errc::address_in_use},
+        {WSAEADDRNOTAVAIL, errc::address_not_available},
         {WSAENETDOWN, errc::network_down},
-        {WSAENETRESET, errc::network_reset},
         {WSAENETUNREACH, errc::network_unreachable},
+        {WSAENETRESET, errc::network_reset},
+        {WSAECONNABORTED, errc::connection_aborted},
+        {WSAECONNRESET, errc::connection_reset},
         {WSAENOBUFS, errc::no_buffer_space},
-        {WSAENOPROTOOPT, errc::no_protocol_option},
+        {WSAEISCONN, errc::already_connected},
         {WSAENOTCONN, errc::not_connected},
-        {WSAENOTSOCK, errc::not_a_socket},
-        {WSAEOPNOTSUPP, errc::operation_not_supported},
-        {WSAEPROTONOSUPPORT, errc::protocol_not_supported},
-        {WSAEPROTOTYPE, errc::wrong_protocol_type},
         {WSAETIMEDOUT, errc::timed_out},
-        {WSAEWOULDBLOCK, errc::operation_would_block},
+        {WSAECONNREFUSED, errc::connection_refused},
+        {WSAENAMETOOLONG, errc::filename_too_long},
+        {WSAEHOSTUNREACH, errc::host_unreachable},
     };
 
+    [[nodiscard]] constexpr bool _Windows_errors_are_sorted_and_unique() {
+        for (size_t _Idx = 1; _Idx < _STD size(_Win_errtab); ++_Idx) {
+            if (_Win_errtab[_Idx - 1]._Windows >= _Win_errtab[_Idx]._Windows) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    static_assert(_Windows_errors_are_sorted_and_unique(),

```

Comment:
```suggestion
    static_assert(_RANGES adjacent_find(_Win_errtab, _RANGES greater_equal{}, &_Win_errtab_t::_Windows) == _STD end(),
```
(Don't forget to `#include<algorithm>`.)

Reply:
Unfortunately, this is built as `/clr:pure` internally, see `src/vctools/crt/managed/msvcurt.settings.targets`. Therefore we can't use C++20 Ranges.

I was slightly worried about including `<algorithm>`, but this lives in the STL's DLL (not import lib), so I think we can do that.

Reply:
`#ifndef _M_CEE_PURE` is about the same number of lines as the lambda, so I see no reason to prefer Ranges or non-Ranges. Pick =)

Reply:
Verified that `adjacent_find` detects both duplication and transposition.

Reply:
Ah, I didn't think about `#ifndef _M_CEE_PURE`! I'll pick plain `_STD` since it's fewer editor lines (ranges would save 1, but cost `#include <functional>`, `#ifndef`, `#endif`).

Reply:
D'oh - `adjacent_find` is `constexpr` in C++20, so we need the ifdef.